lib/ch/query.ex

```diff
@@ -252,51 +252,12 @@ defimpl DBConnection.Query, for: Ch.Query do
     [encode_array_param(k), ?:, encode_array_param(v)]
   end
 
-  @compile inline: [escape_array_param: 1]
-  defp escape_array_param(s) do
-    escape_array_param(s, 0, s, [])
+  defp escape_param([{pattern, replacement} | escapes], param) do
+    param = String.replace(param, pattern, replacement)
+    escape_param(escapes, param)
   end
 
-  @dialyzer {:no_improper_lists, escape_array_param: 4, escape_array_param: 5}
-
-  @doc false
-  # based on based on https://github.com/elixir-plug/plug/blob/main/lib/plug/html.ex#L41-L80
-  def escape_array_param(binary, skip, original, acc)
```

Comment:
It's probably not important for us to be super efficient in query param encoding since they are usually small and few.